app/controllers/api/documents_controller.rb

```diff
@@ -16,11 +16,20 @@ def document_type
       params[:document_type]
     end
 
+    def keyword
+      params[:keyword]
+    end
+
     def filter_documents
       if document_type.present?
         @documents = @documents.joins(:layout)
           .where('comfy_cms_layouts.identifier' => document_type)
       end
+
+      if keyword.present?
+        @documents = @documents.with_content_like(keyword)
```

Comment:
I'm not sure what kind of load Fincap is looking at supporting. It might be worth considering load- testing under load, with a full database.